hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -1236,18 +1248,19 @@ private HFileContext createFileContext(Compression.Algorithm compression,
                                 .withIncludesMvcc(includeMVCCReadpoint)
                                 .withIncludesTags(includesTag)
                                 .withCompression(compression)
-                                .withCompressTags(family.isCompressTags())
-                                .withChecksumType(checksumType)
-                                .withBytesPerCheckSum(bytesPerChecksum)
+                                .withCompressTags(getColumnFamilyDescriptor().isCompressTags())
```

Comment:
Does it simplify things to be able to pass StoreContext directly to the builders?

Reply:
I'd like to keep the current way that passing required fields to the creation of `HFileContext`.  Mainly `HFileContext` is also a member of the `StoreContext` which if we pass into `StoreContext` to the creation of `HFileContext`, it will be a loop and is very strange.